api/handlers/v1/resource.go

```diff
@@ -113,6 +113,9 @@ func (server APIServer) ListResources(ctx context.Context, request *entropyv1bet
 		return nil, status.Error(codes.Internal, "failed to fetch resources from db")
 	}
 	for _, res := range resources {
+		if res.IsDeleted {
```

Comment:
@ishanarya0 database itself should filter out soft-deleted resources and not the handlers,